src/NzbDrone.Core/MediaFiles/EpisodeFileMovingService.cs

```diff
@@ -129,17 +129,18 @@ private EpisodeFile TransferFile(EpisodeFile episodeFile, Series series, List<Ep
             {
                 var scriptImportDecision = _scriptImportDecider.TryImport(episodeFilePath, destinationFilePath, localEpisode, episodeFile, mode);
 
-                switch (scriptImportDecision)
+                transfer &= scriptImportDecision == ScriptImportDecision.DeferMove;
```

Comment:
Since `transfer` is only used with `DeferMove` I think this could be simplified to:

```
if (scriptImportDecision == ScriptImportDecision.DeferMove)
{
    _diskTransferService.TransferFile(episodeFilePath, destinationFilePath, mode);
}
else if (scriptImportDecision == ScriptImportDecision.RenameRequested)
{
    try
    {
        MoveEpisodeFile(episodeFile, series, episodeFile.Episodes);
    }
    catch (SameFilenameException)
    {
        _logger.Debug("No rename was required. File already exists at destination.");
    }
}
```

Reply:
Well not quite it's also used when local episode is null. I didn't want to repeat the transfer call. That's why it's structured like this.

Reply:
Actually thought up a solution that should be ok with pattern matching. This should be ok now.

Reply:
:( I was completely ignoring that case, but this change looks reasonable.